lib/webauthn/base_validation_service.rb

```diff
@@ -31,9 +32,26 @@ def validate_rp_id_hash
       )
     end
 
-    def validate_user_verification
+    ## flags per specification
+    # https://www.w3.org/TR/webauthn-2/#sctn-authenticator-data
+    # bit 0 - user presence
+    # bit 1 - reserved for future use
+    # bit 2 - user verification
+    # bit 3-5 - reserved for future use
+    # bit 6 - attested credential data
+    # bit 7 - extension data
+
+    def validate_user_presence
```

Comment:
Previously, we were checking for the user presence flag only but the validation was using the wrong name. The change here adds the correct error message for user presence and adds the correct flow for user verification. (Note that in the registration/authentication services, verification is only used for first-factor keys.)